cpp/tests/test_abm.cpp

```diff
@@ -238,63 +246,173 @@ TEST(TestLocation, beginStep)
+        params.set<mio::abm::SevereToRecovered>({{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::SevereToRecovered>()[{age, vaccination_state}] = 0.5;
+        params.set<mio::abm::CriticalToDead>({{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::CriticalToDead>()[{age, vaccination_state}] = 0.5;
+        params.set<mio::abm::CriticalToRecovered>({{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::CriticalToRecovered>()[{age, vaccination_state}] = 0.5;
+        params.set<mio::abm::RecoveredToSusceptible>(
+            {{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::RecoveredToSusceptible>()[{age, vaccination_state}] = 0.5;
+        params.set<mio::abm::SusceptibleToExposedByCarrier>(
+            {{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::SusceptibleToExposedByCarrier>()[{age, vaccination_state}] = 0.4;
+        params.set<mio::abm::SusceptibleToExposedByInfected>(
+            {{mio::abm::AgeGroup::Count, mio::abm::VaccinationState::Count}, 0.});
+        params.get<mio::abm::SusceptibleToExposedByInfected>()[{age, vaccination_state}] = 0.5;
+
+        //setup location with some chance of exposure
+        auto home      = mio::abm::Location(mio::abm::LocationType::Home, 0, 0);
+        auto location1 = mio::abm::Location(mio::abm::LocationType::PublicTransport, 0, 3);
+        auto infected1 = mio::abm::Person(home, mio::abm::InfectionState::Carrier, mio::abm::AgeGroup::Age15to34,
+                                          params, vaccination_state);
+        home.add_person(infected1);
+        infected1.migrate_to(home, location1, {0});
+        auto infected2 = mio::abm::Person(home, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age80plus,
+                                          params, vaccination_state);
+        home.add_person(infected2);
+        infected2.migrate_to(home, location1, {0, 1});
+        auto infected3 = mio::abm::Person(home, mio::abm::InfectionState::Infected, mio::abm::AgeGroup::Age5to14,
+                                          params, vaccination_state);
+        home.add_person(infected3);
+        infected3.migrate_to(home, location1, {1});
+
+        //cache precomputed results
+        auto dt                = mio::abm::seconds(8640);
+        bool consider_capacity = false;
+        location1.begin_step(dt, params, consider_capacity);
+
+        ASSERT_TRUE(std::abs(location1.get_cells()[0].cached_exposure_rate[{age, vaccination_state}] - 0.9) < 0.001);
+        ASSERT_TRUE(std::abs(location1.get_cells()[1].cached_exposure_rate[{age, vaccination_state}] - 1) < 0.001);
+        ASSERT_TRUE(std::abs(location1.get_cells()[2].cached_exposure_rate[{age, vaccination_state}]) < 0.001);
+    }
```

Comment:
There is nothing new here, is it? The formatting is very weird.

Reply:
no